provenance-core/src/main/java/ca/on/oicr/gsi/provenance/DefaultProvenanceClient.java

```diff
@@ -443,4 +443,17 @@ public String toString() {
         }
     }
 
+	@Override
+	public void close() throws Exception {
+		Stream.<Collection<? extends AutoCloseable>>of(sampleProvenanceProviders.values(),
+				laneProvenanceProviders.values(), analysisProvenanceProviders.values()).flatMap(Collection::stream)
+				.forEach(c -> {
+					try {
+						c.close();
+					} catch (Exception e) {
+						e.printStackTrace();
```

Comment:
I don't like it, but don't have a better suggestion since

* you're trying to close multiple things, so throwing the exception immediately leaves the others all open
* you could collect and rethrow multiple exceptions inside of one "superexception," but that's ugly too
* exceptions on closing resources are rarely handled well, and I'm not entirely sure what handling them well even looks like
* it never happens anyway

:man_shrugging:

Reply:
Okay. I've modified this to collect all the exceptions that are thrown, pick one, add all the other exceptions as suppressed exceptions and throw the first one.